components/web_discovery/browser/privacy_guard.cc

```diff
@@ -0,0 +1,258 @@
+      VLOG(1) << "Ignoring query due to long word";
+      return true;
+    }
+  }
+  if (query.length() >= kHashCheckMinimumLength) {
+    if (IsHashLikely(query, kHashCheckThresholdMultiplier)) {
+      VLOG(1) << "Ignoring query due to potential inclusion of hash";
+      return true;
+    }
+  }
+  return false;
+}
+
+GURL GeneratePrivateSearchURL(const GURL& original_url,
+                              const std::string& query,
+                              const PatternsURLDetails& matching_url_details) {
+  url::RawCanonOutputT<char> query_encoded;
+  url::EncodeURIComponent(query, &query_encoded);
+  std::string query_encoded_str(query_encoded.view());
+  base::ReplaceSubstringsAfterOffset(&query_encoded_str, 0, "%20", "+");
+
+  return GURL(
+      base::StrCat({original_url.scheme(), url::kStandardSchemeSeparator,
+                    original_url.host(), "/",
+                    matching_url_details.search_template_prefix.value_or(
+                        kDefaultSearchPrefix),
+                    query_encoded_str}));
+}
+
+bool ShouldDropLongURL(const GURL& url) {
+  if (g_regex_util.CheckForEmail(url.spec())) {
+    return true;
+  }
+  if (!url.query_piece().empty()) {
+    if (url.query_piece().size() > kMaxQueryStringLength) {
+      return true;
+    }
+    auto query_parts = base::SplitString(
+        url.query_piece(), "&;", base::WhitespaceHandling::KEEP_WHITESPACE,
+        base::SplitResult::SPLIT_WANT_ALL);
+    if (query_parts.size() > kMaxQueryStringParts) {
+      return true;
+    }
+    if (g_regex_util.CheckForLongNumber(url.query_piece(),
+                                        kMaxQueryStringOrPathNumberLength)) {
+      return true;
+    }
+  }
+  if (!url.path_piece().empty()) {
+    if (g_regex_util.CheckForLongNumber(url.path_piece(),
+                                        kMaxQueryStringOrPathNumberLength)) {
+      return true;
+    }
+  }
+  auto path_parts = base::SplitString(url.path_piece(), "/._ -:+;",
+                                      base::WhitespaceHandling::KEEP_WHITESPACE,
+                                      base::SPLIT_WANT_ALL);
+  for (const auto& path_part : path_parts) {
+    if (path_part.length() > kMaxPathPartLength) {
+      return true;
+    }
+    if (path_part.length() >= kMinPathPartHashCheckLength &&
+        IsHashLikely(path_part)) {
+      return true;
+    }
+  }
+  auto path_segments = base::SplitString(
+      url.path_piece(), "/", base::WhitespaceHandling::KEEP_WHITESPACE,
+      base::SPLIT_WANT_ALL);
+  for (const auto& path_segment : path_segments) {
+    std::string alphanumeric_path_segment = path_segment;
+    TransformToAlphanumeric(alphanumeric_path_segment);
+    if (alphanumeric_path_segment.length() >= kMinSegmentHashCheckLength &&
+        IsHashLikely(alphanumeric_path_segment)) {
+      return true;
+    }
+  }
+  return ContainsForbiddenKeywords(url);
+}
+
+std::optional<std::string> MaskURL(const GURL& url) {
```

Comment:
i would add a comment

Reply:
for what, exactly?

Reply:
describing what is "masking", it's not really clear what the function does and why there is hardcoded something about google inside

Reply:
sorry i meant to expand the comment for this function in the header